generator/uscore/uscore_generator.rb

```diff
@@ -217,7 +217,7 @@ def create_revinclude_test(sequence)
                 assert_response_ok(reply)
                 assert_bundle_response(reply)
                 #{resource_variable} = reply&.resource&.entry&.map(&:resource)&.any? { |resource| resource.resourceType == '#{resource_name}' }
-                assert #{resource_variable}, 'No #{resource_name} resources were returned from this search'
+                skip 'No #{resource_name} resources were returned from this search' unless #{resource_variable}
```

Comment:
I think this was correct as it was. This change skips the test if no Provenance resources are returned, not if none of the resources the Provenance resources should refer to are found. The search and interaction tests skip using this when no resources have been found:
```
skip 'No resources appear to be available for this patient. Please use patients with more information.' unless @resources_found
```

Reply:
revInclude is weird and a little mind-bendy, but in practice for US Core, using AllergyIntolerance (as an example) we should:
1) skip this test if no AllergyIntolerances exist for the patient.  I agree that skip line you provided would accomplish this, but I'm not seeing that line in the test.  And my update doesn't directly test this, but instead assumes that if you don't have any provenance, then you won't have any AlergyIntolerance, and then it skips.
2) skip this test if no Provenances are linked to any AllergyIntollerances for this patient.  We need to know they support provenances for AllergyIntolerances, so skip if we don't see any.

I could be missing something big, obviously.

Reply:
Yeah, I think we need two skips in order for the skips to be clear and not misleading:

- One if no resources of this type have been found at all (could use the line I quoted)
- One if no Provenance resources are found.

Reply:
Yeah, I think we need two skips in order for the skips to be clear and not misleading:

- One if no resources of this type have been found at all (could use the line I quoted)
- One if no Provenance resources are found.